daemon/src/projection.rs

```diff
@@ -208,18 +220,76 @@ pub struct Cfd {
 /// - It serves as an aggregate that is hydrated from events.
 ///
 /// This dual-role motivates the existence of this struct.
-#[derive(Default, Clone, Debug)]
+#[derive(Clone, Debug)]
 struct Aggregated {
     /// If this is present, we have an active DLC.
     latest_dlc: Option<Dlc>,
     /// If this is present, it should have been published.
     collab_settlement_tx: Option<(Transaction, Script)>,
     /// If this is present, it should have been published.
     cet: Option<Transaction>,
-    closing_price: Option<Price>,
+
+    /// If this is present the cet has not been published
+    timelocked_cet: Option<Transaction>,
 
     commit_published: bool,
     refund_published: bool,
+
+    role: Role,
+}
+
+impl Aggregated {
+    fn new(role: Role) -> Self {
+        Self {
+            latest_dlc: None,
+            collab_settlement_tx: None,
+            cet: None,
+            timelocked_cet: None,
+            commit_published: false,
+            refund_published: false,
+            role,
+        }
+    }
+
+    fn payout(self) -> Option<Amount> {
```

Comment:
Unfortunately it's not that simple, because in `apply` I don't know the role. We don't store the role in the `Cfd`. I could add it to the `Cfd` and avoid serialization, but then it's probably better to just keep it in the `Aggregate`.